### PR TITLE
Allow filtering files and programs when showing verification statistics

### PR DESCRIPTION
**Pull request by usamasaqib**
### What does this PR do?
This PR allows a user to pass a file filter and a program filter to selectively show the verification statistics of eBPF programs.

### Motivation
Improves feedback time and eases analysis.

### CODE REVIEW

**Comment by amitslavin** [comment]
Nice @usamasaqib ! Thanks for that!

File: pkg/ebpf/verifier/calculator/main.go [hunk 1]
```diff
@@ -53,6 +70,18 @@ func main() {
...
 		}
 
+		if len(filterFiles) > 0 {
+			found := false
+			for _, f := range filterFiles {
+				if strings.TrimSuffix(d.Name(), ".o") == f {
+					found = true
+				}
+			}
+			if !found {
```
Comment (brycekahle) [inline pkg/ebpf/verifier/calculator/main.go:80]: This only guarantees that one of the filter files is valid, not that all of them are.

File: pkg/ebpf/verifier/calculator/main.go [hunk 2]
```diff
@@ -53,6 +70,18 @@ func main() {
 			return nil
 		}
 
+		if len(filterFiles) > 0 {
+			found := false
+			for _, f := range filterFiles {
+				if strings.TrimSuffix(d.Name(), ".o") == f {
```
Comment (brycekahle) [inline pkg/ebpf/verifier/calculator/main.go:76]: The current way would match a non-`.o` file if it matched the filter. If you intend to only match `.o` files, invert the match:
```suggestion
				if d.Name() == fmt.Sprintf("%s.o", f) {
```

Comment (brycekahle) [reply pkg/ebpf/verifier/calculator/main.go:76]: Or put the `.o` filter before this loop.

File: pkg/ebpf/verifier/stats.go [hunk 3]
```diff
@@ -145,6 +145,19 @@ func generateLoadFunction(file string, stats map[string]*Statistics, failedToLoa
 			strings.Split(filepath.Base(file), ".")[0], "-", "_",
 		)
 		for _, progSpec := range collectionSpec.Programs {
+
+			if len(filterPrograms) > 0 {
+				found := false
+				for _, filter := range filterPrograms {
+					if filter.FindString(progSpec.Name) != "" {
+						found = true
```
Comment (brycekahle) [inline pkg/ebpf/verifier/stats.go:153]:
```suggestion
						found = true
						break
```

File: tasks/ebpf.py [hunk 4]
```diff
@@ -91,19 +91,37 @@ def cleanup_verifier_stats(verifier_stats):
...
     ctx.run("go build -tags linux_bpf pkg/ebpf/verifier/calculator/main.go")
 
-    debug = "--debug" if debug_build else ""
     env = {"DD_SYSTEM_PROBE_BPF_DIR": "./pkg/ebpf/bytecode/build"}
-    res = ctx.run(f"{sudo} ./main {debug}", env=env, hide='out')
+    args = (
+        [
+            "--debug" if debug_build else "",
+        ]
+        + [f"-filter-file {f.split('.')[0]}" for f in filter_file]
```
Comment (brycekahle) [inline tasks/ebpf.py:121]: This way will trim only the last `.ext`:
```suggestion
        + [f"-filter-file {os.path.splitext(f)[0]}" for f in filter_file]
```

File: tasks/ebpf.py [hunk 5]
```diff
@@ -91,19 +91,37 @@ def cleanup_verifier_stats(verifier_stats):
...
         build_cws_object_files(ctx)
 
     ctx.run("go build -tags linux_bpf pkg/ebpf/verifier/calculator/main.go")
 
-    debug = "--debug" if debug_build else ""
     env = {"DD_SYSTEM_PROBE_BPF_DIR": "./pkg/ebpf/bytecode/build"}
-    res = ctx.run(f"{sudo} ./main {debug}", env=env, hide='out')
+    args = (
+        [
+            "--debug" if debug_build else "",
```
Comment (brycekahle) [inline tasks/ebpf.py:119]: mixing double-dash and single-dash args. Let's be consistent.

**Review by brycekahle — CHANGES_REQUESTED** [review]
Some minor comments

File: pkg/ebpf/verifier/calculator/main.go (hunk 2 above)
Comment (usamasaqib) [reply pkg/ebpf/verifier/calculator/main.go:76]: Removing any operation on the filter. Will use it as the user provided.

File: tasks/ebpf.py (hunk 4 above)
Comment (usamasaqib) [reply tasks/ebpf.py:121]: Removing this and adding a separate loop to verify that the extension is valid, i.e. `.o`

**Review by brycekahle — APPROVED** [review]

**Pull request merged**